### PR TITLE
Catch runtime errors

### LINKED ISSUE #11: Catch when service calls fail with runtime errors

### PR DESCRIPTION
**Pull request by khalidharun**
Catches runtime errors, which resolves #11 

Also exporting a `generate_hash` function.

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -1,6 +1,7 @@
-# Generated by roxygen2 (4.1.1): do not edit by hand
+# Generated by roxygen2 (4.1.0): do not edit by hand
```
Comment (peterhurford) [inline NAMESPACE]: Upgrade your Roxygen :+1:

File: tests/testthat/test-servicetests.R [hunk 2]
```diff
@@ -157,6 +157,15 @@ describe('when a service is registered', {
...
   })
 
+  it('fails when service call has a runtime error', {
+    fn1 <- function(x,y) { stop("beep beep. This is just a test.") }
+
+    expect_false(test_service(obj))
+
+    rm("fn1")
+    expect_true(test_service(obj))
+  })
```
Comment (peterhurford) [inline tests/testthat/test-servicetests.R]: Nice :)

File: NAMESPACE (hunk 1 above)
Comment (khalidharun) [inline NAMESPACE]: good point.

**Comment by khalidharun** [comment]
@robertzk @peterhurford Can one you review and merge plz? Thanks.

File: R/servicetests.R [hunk 3]
```diff
@@ -74,17 +82,26 @@ register_service_calls <- function(call_list, envir = parent.frame(2)) {
...
-  if (!is.null(on.pass) && !is.function(on.pass)) stop("on.pass must be a function.")
-  response <- do.call(eval(parse(text = obj$fname), envir = envir), obj$params, envir = envir)
-  passed <- isTRUE(all.equal(response, obj$response))
+#' @param on.pass_each function. Callback function when a test passes.
+#' @param on.fail_each function. Callback function when a test fails.
+#' @param verbose logical. Display messages as each call is processed.
+test_service <- function(obj, envir = parent.frame(), on.pass_each = NULL, on.fail_each = NULL, verbose = TRUE) {
+  if (!is.null(on.fail_each) && !is.function(on.fail_each)) stop("on.fail_each must be a function.")
+  if (!is.null(on.pass_each) && !is.function(on.pass_each)) stop("on.pass_each must be a function.")
+  if (verbose) message(productivus::pp("Calling #{obj$fname}(#{paste(obj$params, collapse = ', ')}) ... "), appendLF = FALSE)
```
Comment (peterhurford) [inline R/servicetests.R]: Just use `paste` and you won't need the `productivus` package anymore.

Comment (khalidharun) [inline R/servicetests.R]: Good point.. I'm getting rid of the verbose option entirely.  That's what the on.pass_each and on.fail_each were designed to do.

**Comment by khalidharun** [comment]
Killing this PR and going with #15 instead.

**Pull request closed without merging**